experimental/framework/fluid-static/src/FluidStatic.ts

```diff
@@ -4,64 +4,129 @@
  */
 
 import { getContainer, IGetContainerService } from "@fluid-experimental/get-container";
-import { getObjectWithIdFromContainer } from "@fluidframework/aqueduct";
+import { DataObject, getObjectWithIdFromContainer } from "@fluidframework/aqueduct";
 import { IContainer } from "@fluidframework/container-definitions";
-import { IFluidDataStoreFactory, NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
-import { DOProviderContainerRuntimeFactory } from "./containerCode";
+import { NamedFluidDataStoreRegistryEntry } from "@fluidframework/runtime-definitions";
+import { DOProviderContainerRuntimeFactory, IFluidStaticDataObjectClass } from "./containerCode";
 
-export interface IFluidStaticDataObjectClass {
-    readonly factory: IFluidDataStoreFactory;
+export interface FluidContainerConfig {
+    id: string,
+    dataObjects: IFluidStaticDataObjectClass[],
+}
+
+export interface FluidCreateContainerConfig extends FluidContainerConfig {
+    /**
+     * initial DataObjects will be created when the Container is first created.
+     */
+    initialDataObjects?: [string, IFluidStaticDataObjectClass][]
 }
 
 export class FluidContainer {
-    constructor(private readonly container: IContainer, public readonly createNew: boolean) { }
+    private readonly types: Set<string>;
```

Comment:
I don't like that this duplicates the registry stored by the `ContainerRuntime` -- the runtime already throws an appropriate error message if the user tries to create an unregistered DO, right?

Reply:
The ContainerRuntime will throw an error if you try to create a DO that is not registered. I could update some handling to do a try/catch for the create instance. 

However, it doesn't validate that there are not duplicate mapping of type to factory in the array. Instead it just chooses the first one. We could update this but historically it hasn't been a big deal because the developer has been defining the type. 

In this case we are telling developers to pull in DataObjects and which come with their own type. We should probably just push this logic down though.